src/elm/arithmetic.ts

```diff
@@ -391,9 +392,13 @@ export class MinValue extends Expression {
 
   valueType: keyof typeof MinValue.MIN_VALUES;
 
-  constructor(json: any) {
+  constructor(json: ELM.MinValue) {
     super(json);
-    this.valueType = json.valueType;
+    if (json.valueType in MinValue.MIN_VALUES) {
+      this.valueType = json.valueType as keyof typeof MinValue.MIN_VALUES;
+    } else {
+      throw new Error(`Not expecting MIN_VALUE: ${json.valueType})`);
```

Comment:
This error check breaks the arithmetic tests, but for a very logical reason lol. There is a [test case](https://github.com/cqframework/cql-execution/blob/3b667a6f0cca2fd6d1b4772f9457f70325e866d0/test/elm/arithmetic/arithmetic-test.ts#L267) that asserts that any incorrect `MinValue` provided should throw an error _during execution_. The generated ELM for this test case is [here](https://github.com/cqframework/cql-execution/blob/3b667a6f0cca2fd6d1b4772f9457f70325e866d0/test/elm/arithmetic/data.js#L2799).

Since the constructor now throws an error if provided a `valueType` that we don't recognize.

I'm going to defer to @cmoesel here on how exactly to proceed. I think I might lean towards not throwing an error in a constructor, but I see the merit in the code that you've added.

Reply:
Yeah def agree with you, both ways have merit. Here is a [StackOverflow](https://stackoverflow.com/a/77705) discussing. I semi-lean towards validation in the constructor as long as that runs over the whole ELM tree as soon as the ELM wrapper is instantiated, just so that it doesn't become possible for execution to 'skip' certain parts of the ELM tree depending on the input patient data (only exec MinValue node if Patient has Condition Y) which could then allow bugs to escape tests more easily. Not sure if that's applicable though?

Either way, MIN_VALUES could be an ENUM but SHOULD NOT be added to the ELM generated from the XML because the mappings to `MathUtil.MIN_INT_VALUE` seem to be numbers that are only specific to the JS implementation.

Reply:
I can see both arguments -- including the desire to fail immediately upon loading the ELM (rather than on execution, just in case execution never occurs).  That said the spec does say this:
> For any other type, attempting to invoke minimum results in an error.

One could argue that "attempting to invoke" implies an error on execution, not on loading. (As far as I can tell, this is how the [Java cql-engine implementation](https://github.com/DBCG/cql_engine/blob/master/engine/src/main/java/org/opencds/cqf/cql/engine/elm/execution/MinValueEvaluator.java) behaves as well).